pkg/middleware/middleware.go

```diff
@@ -39,244 +20,7 @@ var (
-	}
-
-	authQuery := models.LoginUserQuery{
-		Username: username,
-		Password: password,
-	}
-	if err := bus.Dispatch(&authQuery); err != nil {
-		ctx.Logger.Debug(
-			"Failed to authorize the user",
-			"username", username,
-			"err", err,
-		)
-
-		if errors.Is(err, models.ErrUserNotFound) {
-			err = login.ErrInvalidCredentials
-		}
-		ctx.JsonApiErr(401, errStringInvalidUsernamePassword, err)
-		return true
-	}
-
-	user := authQuery.User
-
-	query := models.GetSignedInUserQuery{UserId: user.Id, OrgId: orgId}
-	if err := bus.Dispatch(&query); err != nil {
-		ctx.Logger.Error(
-			"Failed at user signed in",
-			"id", user.Id,
-			"org", orgId,
-		)
-		ctx.JsonApiErr(401, errStringInvalidUsernamePassword, err)
-		return true
-	}
-
-	ctx.SignedInUser = query.Result
-	ctx.IsSignedIn = true
-	return true
-}
-
-func initContextWithToken(authTokenService models.UserTokenService, ctx *models.ReqContext, orgID int64) bool {
-	if setting.LoginCookieName == "" {
-		return false
-	}
-
-	rawToken := ctx.GetCookie(setting.LoginCookieName)
-	if rawToken == "" {
-		return false
-	}
-
-	token, err := authTokenService.LookupToken(ctx.Req.Context(), rawToken)
-	if err != nil {
-		ctx.Logger.Error("Failed to look up user based on cookie", "error", err)
-		WriteSessionCookie(ctx, "", -1)
-		return false
-	}
-
-	query := models.GetSignedInUserQuery{UserId: token.UserId, OrgId: orgID}
-	if err := bus.Dispatch(&query); err != nil {
-		ctx.Logger.Error("Failed to get user with id", "userId", token.UserId, "error", err)
-		return false
-	}
-
-	ctx.SignedInUser = query.Result
-	ctx.IsSignedIn = true
-	ctx.UserToken = token
-
-	// Rotate the token just before we write response headers to ensure there is no delay between
-	// the new token being generated and the client receiving it.
-	ctx.Resp.Before(rotateEndOfRequestFunc(ctx, authTokenService, token))
-
-	return true
-}
-
-func rotateEndOfRequestFunc(ctx *models.ReqContext, authTokenService models.UserTokenService, token *models.UserToken) macaron.BeforeFunc {
-	return func(w macaron.ResponseWriter) {
-		// if response has already been written, skip.
-		if w.Written() {
-			return
-		}
-
-		// if the request is cancelled by the client we should not try
-		// to rotate the token since the client would not accept any result.
-		if errors.Is(ctx.Context.Req.Context().Err(), context.Canceled) {
-			return
-		}
-
-		addr := ctx.RemoteAddr()
-		ip, err := network.GetIPFromAddress(addr)
-		if err != nil {
-			ctx.Logger.Debug("Failed to get client IP address", "addr", addr, "err", err)
-			ip = nil
-		}
-		rotated, err := authTokenService.TryRotateToken(ctx.Req.Context(), token, ip, ctx.Req.UserAgent())
-		if err != nil {
-			ctx.Logger.Error("Failed to rotate token", "error", err)
-			return
-		}
-
-		if rotated {
-			WriteSessionCookie(ctx, token.UnhashedToken, setting.LoginMaxLifetime)
-		}
-	}
-}
-
-func AddDefaultResponseHeaders() macaron.Handler {
+func AddDefaultResponseHeaders(cfg *setting.Cfg) macaron.Handler {
```

Comment:
Is it just me or now that this file is basically gutted, can it be renamed to something else more fitting? Or move these functions to somewhere that already exists - perhaps `/pkg/api/http_server.go` makes sense (but could be wrong)

Reply:
@wbrowne I dunno, not sure what else to call it either. I don't think it really matters all in all. It's often useful though to have a file named after the package, to hold bits that are common to the package, for example its documentation. As such we might as well keep the file.

Reply:
Yeah I think the file definitely makes sense to have - I guess in its current form it's strange to only find 1 public method there  🤷‍♂️  This is only the first step anyway and stuff will likely change, just thought it's worth pointing out now in case you saw somewhere else this code belonged